src/dependencies.js

```diff
@@ -0,0 +1,180 @@
+const { dirname } = require('path')
+
+const precinct = require('precinct')
+const resolve = require('resolve')
+const requirePackageName = require('require-package-name')
+const promisify = require('util.promisify')
+const glob = require('glob')
+const debug = require('debug')('@netlify/zip-it-and-ship-it:finders')
+
+const pResolve = promisify(resolve)
+const pGlob = promisify(glob)
+
+// Retrieve all the files recursively required by a Node.js file
+const getDependencies = async function(handler, packageRoot) {
+  const packageJson = getPackageJson(packageRoot)
+
+  const state = { localFiles: [], modulePaths: [] }
+
+  try {
+    return await getFileDependencies(handler, packageJson, state)
+  } catch (error) {
+    error.message = `In file "${handler}": ${error.message}`
+    throw error
+  }
+}
+
+const getPackageJson = function(packageRoot) {
+  if (packageRoot === undefined) {
+    return {}
+  }
+
+  return require(`${packageRoot}/package.json`)
+}
+
+const getFileDependencies = async function(path, packageJson, state) {
+  if (state.localFiles.includes(path)) {
+    return []
+  }
+
+  state.localFiles.push(path)
+
+  const basedir = dirname(path)
+  // This parses JavaScript in `path` to retrieve all the `require()` statements
+  // TODO: `precinct.paperwork()` uses `fs.readFileSync()` under the hood,
+  // but should use `fs.readFile()` instead
+  const dependencies = precinct.paperwork(path, { includeCore: false })
+
+  const depsPaths = await Promise.all(
+    dependencies.map(dependency => getImportDependencies(dependency, basedir, packageJson, state))
+  )
+  return [].concat(...depsPaths)
+}
+
+// `require()` statements can be either `require('moduleName')` or
+// `require(path)`
+const getImportDependencies = function(dependency, basedir, packageJson, state) {
+  if (LOCAL_IMPORT_REGEXP.test(dependency)) {
+    return getLocalImportDependencies(dependency, basedir, packageJson, state)
+  }
+
+  return getModuleDependencies(dependency, basedir, state, packageJson)
+}
+
+const LOCAL_IMPORT_REGEXP = /^\.|\//
```

Comment:
I think this might be incorrect, I'm guessing we're trying to match `/path/to/file` or `./path/to/file` for local files? This could also the slash in `@apollo/lambda-server`.

```
// startswith '.' or '/'
const LOCAL_IMPORT_REGEXP = /^(\.|\/)/
```

Reply:
Great catch! Fixed in 858a0ae